### ISSUE TITLE
Transect - input form.  Enable 'Country' species list

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/ABLE#233]
On https://butterfly-monitoring.net/ebms-input-data

current species list options are:
- all species
- species known at this site
- species I have recorded

A further option would be helpful:
- country species list

This would be populated using the taxon-attributes available for the eBMS taxon list

**Comment by Gary-van-Breda** [comment]
Country species lists enabled on the Dev site: configured on Hungarian and Luxembourg.
For this trial, I've set it up to use the relevant taxon presence attribute (we can set up a similar attribute if this is not appropriate) for each of those 2 countries, using "P" and "M" values (this can be configured to different values.)

As noted in #278 I've disabled the Full species list because it is so big (unwieldy?) and has performance issues.

**Comment by DavidRoy** [comment]
@xaviermestdagh could you test this for Luxembourg on https://dev-butterfly-monitoring.pantheonsite.io/walk-data-entry-new

**Comment by DavidRoy** [comment]
@Gary-van-Breda I tested out for Hungary.  Am I misunderstanding how this functionality should work?

This screenshot shows that species 'Aglais ichnusa' is still available on the list, even though this taxa is 'A' for HU.
http://warehouse1.indicia.org.uk/index.php/taxa_taxon_list/edit/432427

![wde](https://user-images.githubusercontent.com/2699478/98685760-af02e580-235f-11eb-8b66-6444d124630f.png)

**Comment by DavidRoy** [comment]
Actually, ignore my comment. I've now noticed the 'Branch species' option.  The species search enables the full species list which is sensible.

Can 'Branch specific' be relabelled as 'Region species list'

**Comment by DavidRoy** [comment]
Rename 'Butterfly life stage' (which appears for both the butterfly and moth section). 
to 'Life stage'.

**Comment by DavidRoy** [comment]
Default life stage to 'Adult' throughout

**Comment by Gary-van-Breda** [comment]
Label changed to "Region specific list"
Default life stage set to "Adult"
I'm assuming you don't want the attribute name changed on the warehouse? In which case we need to wait until the live node number is known before we can change the attribute title

**Comment by xaviermestdagh** [comment]
Here a screenshot of what I see, how can I have the view with live stage, outside transect counts...? 
There are several "UNKNOWN DATA TYPE "" FOR ID: CAPTION:"
The page start with a large error (?) code (second screenshot).  
![image](https://user-images.githubusercontent.com/24294138/98940817-d981a980-24eb-11eb-859e-d4ddfe3f7745.png)
![image](https://user-images.githubusercontent.com/24294138/98941161-70e6fc80-24ec-11eb-8d09-f9f0c32eeea2.png)

**Comment by DavidRoy** [comment]
@Gary-van-Breda I get the same error.  Is this to do with the 'reliability' functionality as these errors only show for Luxembourg I think

**Comment by Gary-van-Breda** [comment]
@DavidRoy : sorry - I had some finger trouble, and put in 1383 instead of 1393 for the reliability attribute ID. Now fixed.  
  
@xaviermestdagh : Currently the live stage and outside transect counts are only active for the Hungarian scheme, not Luxembourg, on the Dev site. Would you like me to activate them for Luxembourg as well?

**Comment by Gary-van-Breda** [comment]
Form modified to allow override to occurrence attribute labels.
Deployed to Dev as part of the new walk data entry form, and config set so 'Butterfly life stage' shown as 'Life stage':
https://dev-butterfly-monitoring.pantheonsite.io/walk-data-entry-new

**Comment by DavidRoy** [comment]
Looks good to me.
@xaviermestdagh can you test again?

**Comment by xaviermestdagh** [comment]
Great for Luxembourg. Under discussion with colleagues whether to switch to "live stage" view like in Hungary or not.
@DavidRoy : is it expected to have live stage and outside transect counts available for transect in the mobile app as well? For Bumblebee, recording queen, worker or male is of importance, could it be part of the values in "Live stage"? Alternatively, adding a field sex for all groups...?!

**Comment by Gary-van-Breda** [comment]
@DavidRoy @xaviermestdagh : Live stage view is live for Hungary only. Do you wish for it to be enabled for Luxembourg? (If so do you want the outside transect count, the comments and/or the occurrence images as well?)

Should I also start setting up the config for the Country species lists? What value(s) should I use for the presence attributes to indicate a taxon should be included in the country list?

**Comment by xaviermestdagh** [comment]
@Gary-van-Breda is there a way I can see the hungarian form? (the dev-butterfly... link here below is not working)

**Comment by JimBacon** [comment]
@DavidRoy The question from @Gary-van-Breda is one for you, I think.

> What value(s) should I use for the presence attributes to indicate a taxon should be included in the country list?

The values (from https://warehouse1.indicia.org.uk/index.php/termlist/edit/816) are as follows:

- A: Absent (never recorded in the respective country or island group or only doubtful records).
- P: Present (native or well-established populations, including alien species).
- P?: Possibly present (recorded but continued presence doubtful).
- M: Regular migrant (species which has no permanent populations).
- I: Irregular vagrant (including temporary or recently established populations).
- Ex: Regionally extinct (though vagrants might be seen occasionally).

You might be generous and say 'all except A' or mean and say 'only P', or anything in between!

**Comment by Gary-van-Breda** [comment]
@DavidRoy : Shall I set the country list up for just one country first so you can check it? if so, which one?

**Comment by DavidRoy** [comment]
@Gary-van-Breda good idea.  Please test using Luxembourg so @xaviermestdagh can confirm the list makes sense?

**Comment by xaviermestdagh** [comment]
> @xaviermestdagh You should be able to see it on https://butterfly-monitoring.net/walk-data-entry?sample_id=18498444 : Bare in mind that this is Live Hungarian data
> 
> Would you rather I set it up for Luxembourg on Test or Dev?

Thanks @Gary-van-Breda, I could have a look at live stage view on live on Hungarian site. 
For butterflies I am not in favor of having this live stage view for Luxembourg because: 
- our protocol targets adult stage only, and having the possibility to record other live stages might lead observers to spend very various duration on field.  
- it will be different from the mobile app.
- observations outside transects are encoded through mobile apps (iObs, ObsMapp, iNaturalist, NaturaList (Biolovision)).

**Comment by xaviermestdagh** [comment]
> @Gary-van-Breda good idea. Please test using Luxembourg so @xaviermestdagh can confirm the list makes sense?

If I understood well, list of species should be filtered for Luxembourg? FYI, I still see species with no potential presence in Luxembourg. But Indeed all values except "A" are worth to keep I believe. 
![image](https://user-images.githubusercontent.com/24294138/173313002-625cccde-1bc7-4129-9431-a78cb7a5f612.png)

**Comment by JimBacon** [comment]
_Colias aurorina_ has a value of 'A' for Luxembourg so, when Gary has finished, it should no longer appear in the list for you @xaviermestdagh. 

@Gary-van-Breda, the list now includes Kenyan species, for which the presence value in European countries will be null. Your query may need to be  filter on `!= 'A' and not null` or use an inner join to exclude taxa with a null presence. Also, only the preferred names have the presence information so, if listing by common name, you might need to join by `preferred_taxa_taxon_list_id`
